client/my-sites/stats/controller.jsx

```diff
@@ -335,8 +334,8 @@ export default {
 		const isValidStartDate =
 			queryOptions.startDate && i18n.moment( queryOptions.startDate ).isValid();
 		const date = isValidStartDate
-			? i18n.moment( queryOptions.startDate )
-			: momentSiteZone.endOf( activeFilter.period );
+			? i18n.moment( queryOptions.startDate ).locale( 'en' )
+			: rangeOfPeriod( activeFilter.period, momentSiteZone.locale( 'en' ) ).startOf;
```

Comment:
Could you explain why we went from `momentSiteZone.endOf( activeFilter.period )` to `rangeOfPeriod( activeFilter.period, momentSiteZone.locale( 'en' ) ).startOf;`? 

It seems like we changed the logic to use the start of the period instead of its end. Does this work as expected with the API? For other stats, too?

Reply:
@jsnmoon I changed it to be consistent with the other controller actions, but you're quite right - it could have unintended side effects.

I did some more tinkering and discovered that it's the `.locale('en')` that ultimately fixed the problem. I've applied that to the original code and all seems well now.